internal/controller/controllers/infraenv_controller.go

```diff
@@ -499,7 +505,12 @@ func (r *InfraEnvReconciler) createInfraEnv(ctx context.Context, log logrus.Fiel
 		createParams.InfraenvCreateParams.StaticNetworkConfig = staticNetworkConfig
 	}
 
-	return r.Installer.RegisterInfraEnvInternal(ctx, key, createParams)
+	mirrorRegistryConfiguration, err := mirrorregistries.ProcessMirrorRegistryConfig(ctx, log, r.Client, infraEnv.Spec.MirrorRegistryRef)
```

Comment:
Do we need to handle somewhere a conflict between ACI and InfraEnv? I.e. do we want to settle a potential mismatch in config?

Reply:
Nope, by @avishayt definition the user is able to set different mirror registry for InfraEnv and ACI

Reply:
I forgot to click submit on my comment yesterday 😆 but I had the same thought! 

Do we handle if user sets different mirror registry for both? Do we merge the mirror registries when they start install?

Reply:
Basically, We don't want to check if the mirror registries are different, as long as they are exists. On failure the user will see the status update.
Now that I'm thinking of it I need to add the InfraEnv status update, it's currently updating only the ACI status